### PR TITLE
Adjust input field height and input label text size

### LINKED ISSUE #49477: [Enhancement]: Field height and label text size adjustments in the Checkout block
### Describe the solution you'd like

With the PR #46362, we've implemented the issue #46008, which we reverted with the PR #49404, after we noticed that under certain conditions checkout input labels and values might be rendered invisible. To ship a point release as fast as possible, we decided to do a full revert. This issue aims to port back the following changes of the PR #46362 into the codebase:

```[tasklist]
### Tasks
- [x] Ensure that text input fields have a consistent height of 50px.
- [x] Update the labels of text fields to use a font size of approximately 12px when the input field contains text (with a scaling factor of 0.75 when text is present).
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nielslange**
### Changes proposed in this Pull Request:

This PR introduces the following changes:

- Ensure that text input fields have a consistent height of 50px.
- Update the labels of text fields to use a font size of approximately 12px when the input field contains text (with a scaling factor of 0.75 when text is present).

Closes #49477.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the frontend.
2. Add a product to the cart.
3. Go to the cart page that contains the Cart block.
4. Verify that the coupon input field has a height of 50px.
5. Go to the checkout page that contains the Checkout block.
6. Verify that all text input fields and the native select fields have a height of 50px.
7. Verify that the text size of the input field labels is approximately 12px when the input field has a value.

<table>
<tr>
<td valign="top">Before:
<br><br>
Screenshot 2024-07-17 at 12 41 16

</td>
<td valign="top">After:
<br><br>
Screenshot 2024-07-17 at 12 36 14

</td>
</tr>
</table>

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Adjust input field height and input label text size.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/base/components/select/style.scss [hunk 1]
```diff
@@ -29,7 +29,7 @@
 		border-radius: $universal-border-radius;
 		border: none;
 		width: 100%;
-		height: 100%;
+		height: 50px;
```
Comment (wavvves) [inline plugins/woocommerce-blocks/assets/js/base/components/select/style.scss]: I see that `height` for the parent container was already set at `50px`, is this really necessary?

**Review by wavvves — COMMENTED** [review]
Hi Niels, thank you for working on this. I just left a tiny question before approving 👍🏼

File: plugins/woocommerce-blocks/assets/js/base/components/select/style.scss (hunk 1 above)
Comment (nielslange) [reply plugins/woocommerce-blocks/assets/js/base/components/select/style.scss]: Good catch, @wavvves. That definition was not necessary, and I reverted my initial change back to `height: 100%` to ensure that the select field uses up the entire height.

**Comment by nielslange** [comment]
> Hi Niels, thank you for working on this. I just left a tiny question before approving 👍🏼

Thanks for your review and your suggestion, @wavvves. I just updated the PR accordingly. You can now review the PR again.

**Review by wavvves — APPROVED** [review]
LGTM! Thanks Niels!

**Comment by nielslange** [comment]
Thanks for reviewing and approving the PR, @wavvves. 🙌

**Pull request merged**